### PR TITLE
Eval

### PR DESCRIPTION
**Pull request by WesleyMPG**
It's not finished, but works. 

The first problem was to stop a regular playback without wait for it finish. I couldn't think anything to do that the way things were with the capture file. So the idea is capture file should be just a text file in which each line is the python command (basically the way it is now, but text file instead python file).
Python has a funtion called eval() that executes the content of a string as python commands. This way commands can be executed in a loop that can be interrupted by a flag.

The second problem was the while starting threads. Atbswp frozes because this while blocks the program execution. To solve that just delegated the while to a thread.

It's important to say that it is not ready to merge yet. Changes are in a tempory state and the code is not as i described yet. The ideia is that and it's working more or less that way, but need to be polished. Most the capture file handling.

### CODE REVIEW

File: atbswp/temp.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (RMPR) [inline atbswp/temp.py:1]: It seems like you accidentally pushed the file your are using for tests.

**Comment by RMPR** [comment]
> It&#39;s not finished, but works.
>
> The first problem was to stop a regular playback without wait for it
> finish. I couldn&#39;t think anything to do that the way things were
> with the capture file. So the idea is capture file should be just a text
> file in which each line is the python command (basically the way it is
> now, but text file instead python file).
> Python has a funtion called eval() that executes the content of a string
> as python commands. This way commands can be executed in a loop that can
> be interrupted by a flag.
>
> The second problem was the while starting threads. Atbswp frozes because
> this while blocks the program execution. To solve that just delegated
> the while to a thread.


Thanks for the patch, it seems to work as advertised. Tbh, 
I had a fix more or less ready for the infinite playback thing, on the 
`develop` branch.
You can `git checkout develop` and have a quick look if it works for 
you, especially this commit[0] and this one[1]. Keep in mind that, even
though it might work, you can still help me spot bugs that might get
overlooked otherwise. Here's[2] the corresponding announcement in the
community and a small write-up about how and why things were implemented
that way, if you are interested. I think we can get to a satisfying
solution working together.

0:
https://github.com/RMPR/atbswp/commit/345de950bed2c8f98a1586fff1f9beed3f76162b


1:
https://github.com/RMPR/atbswp/commit/1861540857302e31042ec36df213728188ca4dc8


2: https://t.me/atbswp/41

File: atbswp/control.py [hunk 2]
```diff
@@ -401,7 +407,7 @@ def action(self, event):
             else:
                 path = os.path.join(Path(__file__).parent.absolute(), 'atbswp.py')
             settings.save_config()
-            os.execl(path)
+            os.execl(path, *sys.argv)
```
Comment (RMPR) [inline atbswp/control.py:448]: This might lead to a crash (see my comment on the PR)

**Review by RMPR — CHANGES_REQUESTED** [review]

File: atbswp/temp.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"pyautogui.moveTo(679, 348)",
+"pyautogui.moveTo(573, 372)",
+"pyautogui.moveTo(569, 394)",
+"time.sleep(1)",
+"pyautogui.moveTo(590, 285)",
+"pyautogui.moveTo(611, 236)",
+"pyautogui.moveTo(632, 213)",
+"pyautogui.moveTo(653, 193)",
+    ]
+
```
Comment (RMPR) [inline atbswp/temp.py:39]: It seems like you accidentally pushed the file your are using for tests.

**Review by RMPR — CHANGES_REQUESTED** [review]

File: atbswp/temp.py (hunk 3 above)
Comment (WesleyMPG) [reply atbswp/temp.py:39]: Sorry for that. I had accidentaly reseted my .gitiginore haha

File: atbswp/control.py (hunk 2 above)
Comment (WesleyMPG) [reply atbswp/control.py:448]: I realized now that i made the changes based on master. And seeing your fix I was like: we did the same thing haha. If you asked me before i would swear that the exec on play function was also some os.exec and the playback was a separated program.

**Comment by RMPR** [comment]
> I realized now that i made the changes based on master. And seeing your
> fix I was like: we did the same thing haha. 

Yeah :) the same idea prevails. By the way, it seems like my changes
introduced a bug with the repeat count (I will upload a GIF). I think it's 
a synchronization bug with threads. I plan to investigate when I
have time, but if you're still interested in working on that feature, 
feel free to spot the error in my code.


> If you asked me before i
> would swear that the exec on play function was also some os.exec and the
> playback was a separated program.

That's why I removed it all together, it didn't make much sense to me at
that point.

**Comment by RMPR** [comment]
atbswp repeat count issue

Sorry for the quality.

**Comment by WesleyMPG** [comment]
First of all, I pressed the Review request by mistake haha. 
And I can give a look for this repeat counter bug.

**Comment by RMPR** [comment]
Closing this PR since it's not useful anymore.

**Pull request closed without merging**